xcdat/dataset.py

```diff
@@ -643,3 +668,43 @@ def _split_time_units_attr(units_attr: str) -> Tuple[str, str]:
         )
 
     return units, reference_date
+
+
+def _get_cftime_coords(
+    ref_date: str, offsets: np.ndarray, calendar: str, units: str
+) -> np.ndarray:
+    """Get an array of `cftime` coordinates starting from a reference date.
+
+    Parameters
+    ----------
+    ref_date : str
+        The starting reference date.
+    offsets : np.ndarray
+        An array of numerically encoded time offsets from the reference date.
+    calendar : str
+        The CF calendar type.
+    units : str
+        The time units.
+
+    Returns
+    -------
+    np.ndarray
+        An array of `cftime` coordinates.
+    """
+    # Starting from the reference date, create an array of `datetime` objects
+    # by adding each offset (a numerically encoded value) to the reference date.
+    # The `parse.parse` default is set to datetime(2000, 1, 1), with each
+    # component being a placeholder if the value does not exist. For example, 1
+    # and 1 are placeholders for month and day if those values don't exist.
+    ref_datetime: datetime = parser.parse(ref_date, default=datetime(2000, 1, 1))
+    offsets = np.array(
+        [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
+        dtype="datetime64",
+    )
+
+    # Convert the array of `datetime`` objects into `cftime` objects based on
+    # the calendar type.
+    date_type = get_date_type(calendar)
+    coords = convert_times(offsets, date_type=date_type)
```

Comment:
I don't mind this. I initially used the `xr` function, but then I wasn't sure if we wanted/needed their `_is_standard_calendar` [check](https://github.com/pydata/xarray/blob/3c98ec7d96cc4b46664850cc7a40af2bc184fea0/xarray/coding/cftime_offsets.py#L73). I guess this will never run, because we set `use_cftime = True`.

Reply:
I think `convert_times` is hitting problems for years far into the future (`convert_times` is based on pandas):

```
import xcdat

fn = '/p/user_pub/climate_work/pochedley1/cmip6_msu/spliced/ttt_Amon_MRI-ESM2-0_historical-ssp585_r1i1p1f1_gn_185001-230012.nc'
ds = xcdat.open_dataset(fn)
```

File ~/code/xc

> dat/xcdat/dataset.py:719, in _get_cftime_coords(ref_date, offsets, calendar, units)
>     716 # Convert the array of `datetime` objects into `cftime` objects based on
>     717 # the calendar type.
>     718 date_type = get_date_type(calendar)
> --> 719 coords = convert_times(offsets, date_type=date_type)
>     721 return coords
> 
> File ~/bin/anaconda3/envs/xcdat_dev/lib/python3.9/site-packages/xarray/coding/times.py:451, in convert_times(times, date_type, raise_on_invalid)
>     448     return cftime_to_nptime(times, raise_on_invalid=raise_on_invalid)
>     449 if is_np_datetime_like(times.dtype):
>     450     # Convert datetime64 objects to Timestamps since those have year, month, day, etc. attributes
> --> 451     times = pd.DatetimeIndex(times)
>     452 new = np.empty(times.shape, dtype="O")
>     453 for i, t in enumerate(times):
> 
> . 
> .
> .
>
> OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 2262-05-01 00:00:00

Reply:
Refer to my suggestion to update the `dtype` of the `offsets` np.ndarray from `datetime64` to `object`.

I validated this fix using your example.